gazebo_ros/test/test_disable_performance_metrics.test.py

```diff
@@ -0,0 +1,172 @@
+
+import unittest
+
+import launch
+import launch.actions
+
+import launch_testing
+import launch_testing.asserts
+import pytest
+import os
+
+import rclpy
+from gazebo_msgs.msg import PerformanceMetrics
+from rclpy.node import Node
+from rcl_interfaces.msg import Parameter, ParameterType
+from rcl_interfaces.srv import SetParameters
+import time
+from threading import Thread
+
+
+@pytest.mark.launch_test
+def generate_test_description():
+    """Launch a gzserver node."""
+    os.chdir(os.path.dirname(__file__))
+    return launch.LaunchDescription([
+        launch.actions.ExecuteProcess(
+            cmd=['gzserver', '-s', 'libgazebo_ros_init.so'],
+            output='screen',
+            sigterm_timeout='20',
+            sigkill_timeout='20'
+        ),
+        launch_testing.actions.ReadyToTest()
+    ])
+
+
+class TestPerformanceMetricsParam_disable(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        rclpy.init()
+
+    @classmethod
+    def tearDownClass(cls):
+        rclpy.shutdown()
+
+    def test_parameter_disable(self):
+        """
+        Test function to set parameter to false.
+
+        Checks if /performance_metrics topic stops publishing if
+        enable_performance_metrics parameter is set to False.
+        """
+        node = MakeTestNode('performance_metrics_disable_testing_node')
+
+        assert node.wait_for_gazebo_node(timeout_sec=5.0),\
+            'Gazebo node not found, parameter could not be set'
+
+        # Set enable_performance_metrics parameter to False and listen for messages
+        response = node.set_parameter(state=False)
+        assert response.successful, 'Parameter could not be set to False'
+
+        node.start_subscriber()
+        time.sleep(2.5)
+        assert node.msg_count == 0, f'Received {node.msg_count} messages after\
+            setting enable_performance_metrics parameter to False, test failed'
+
+
+class TestPerformanceMetricsParam_enable(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        rclpy.init()
+
+    @classmethod
+    def tearDownClass(cls):
+        rclpy.shutdown()
+
+    def test_parameter_enable(self):
+        """
+        Test function to set parameter to true.
+
+        Checks if /performance_metrics topic starts publishing if
+        enable_performance_metrics is set to True.
+        """
+        node = MakeTestNode('performance_metrics_enable_testing_node')
+
+        assert node.wait_for_gazebo_node(timeout_sec=5.0),\
+            'Gazebo node not found, parameter could not be set'
+
+        # Set enable_performance_metrics parameter to True and listen for messages
+        response = node.set_parameter(state=True)
+        assert response.successful, 'Parameter could not be set to True'
+
+        node.start_subscriber()
+        time.sleep(2.5)
+        assert node.msg_count != 0, 'Did not receive any messages after\
+            setting enable_performance_metrics to True, test failed'
```

Comment:
We can do better than blindly sleeping. We could poll with a timeout for `node.msg_count` to be greater than zero. Or even better, use a Python [`Event`](https://docs.python.org/3/library/threading.html#event-objects) object to signal from the spin thread to this thread.

Reply:
Added ``Event()`` based quick return logic